src/main/java/org/jahia/modules/jahiacsrfguard/filters/CsrfGuardJavascriptFilter.java

```diff
@@ -100,10 +119,32 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         }
     }
 
+    private boolean matchHtmlContentType(HttpServletResponse response) {
+        return StringUtils.contains(response.getContentType(),"text/html");
+    }
+
+    private boolean matchUrlResolverPattern(HttpServletRequest httpRequest) {
+        URLResolver urlResolver = (URLResolver) httpRequest.getAttribute("urlResolver");
+        return urlResolver != null  && matchPattern(urlResolver.getPath());
```

Comment:
Does this work when accessing an html/jsp directly (without render service), like in the tools ? urlResolver would be null, and this will skip the filter

Reply:
yes, the condition check is `!(matchPattern(requestPath) || matchUrlResolverPattern(httpRequest))` in case of `jsp` or static `html` file `matchPattern(requestPath)` returns `true`